lib/macho/fat_file.rb

```diff
@@ -346,6 +348,11 @@ def populate_fat_header
       # formats.
       raise JavaClassFileError if fh.nfat_arch > 30
 
+      # Rationale: fat Mach-0 files with no internal slices should
+      # return an error unless the permissive option is set.
+      permissive = options.fetch(:permissive, false)
```

Comment:
IMO we can remove the permissive check here: there's very little a user can do with an empty universal binary, so we should fail unconditionally if we have no architecture slices.

Reply:
That makes sense.